core/src/main/scala/com.snowplowanalytics.stream/loader/Config.scala

```diff
@@ -88,17 +87,19 @@ object Config {
       )
 
       val timestamp = timestampEither.right.toOption
-
-      val endpoint = getKinesisEndpoint(region, customEndpoint)
-
-      val dynamodbEndpoint = dynamodbCustomEndpoint.getOrElse(region match {
-        case cn @ "cn-north-1" => s"https://dynamodb.$cn.amazonaws.com.cn"
-        case _                 => s"https://dynamodb.$region.amazonaws.com"
-      })
     }
 
     object Kinesis {
       final case class Buffer(byteLimit: Long, recordLimit: Long, timeLimit: Long)
+
+      implicit val sourceKinesisConfigReader: ConfigReader[Kinesis] =
+        deriveReader[Kinesis].emap { c =>
+          val region = c.region.orElse(getRegion)
```

Comment:
@istreeter @chuwy Some of the places needs region explicitly therefore I used `DefaultAwsRegionProviderChain` [here](https://github.com/snowplow/snowplow-elasticsearch-loader/pull/203/commits/1069842b1b3ebfd066a2791c3e26cbe7eb84d041#diff-062d73c4a4e2a587b455f5b2a5be3ec957a5406838154fcfa73bf564e811ba94R299) to get the region. However, it creates problem in the tests because it is getting region from local AWS profile. I thought about making the config check in the beginning of the application however in that case copying region as [in this line](https://github.com/snowplow/snowplow-elasticsearch-loader/pull/203/commits/1069842b1b3ebfd066a2791c3e26cbe7eb84d041#diff-062d73c4a4e2a587b455f5b2a5be3ec957a5406838154fcfa73bf564e811ba94R99) will be problem because we will have deeply nested object in the hand. Maybe, we can use lenses but I guess that would be too much just for this task. Another option is delegating usage of the `DefaultAwsRegionProviderChain` to place where `region`s are actually used. Feel free to share any other way if you can think any.

Reply:
The places where region is required:
https://github.com/snowplow/snowplow-elasticsearch-loader/blob/1069842b1b3ebfd066a2791c3e26cbe7eb84d041/core/src/main/scala/com.snowplowanalytics.stream/loader/sinks/KinesisSink.scala#L63

https://github.com/snowplow/snowplow-elasticsearch-loader/blob/1069842b1b3ebfd066a2791c3e26cbe7eb84d041/elasticsearch/src/main/scala/com/snowplowanalytics/stream/loader/clients/SignedHttpClientConfigCallback.scala#L54

https://github.com/snowplow/snowplow-elasticsearch-loader/blob/1069842b1b3ebfd066a2791c3e26cbe7eb84d041/core/src/main/scala/com.snowplowanalytics.stream/loader/executors/KinesisSourceExecutor.scala#L95

Reply:
Hey @spenes I pushed [this commit](https://github.com/snowplow/snowplow-elasticsearch-loader/commit/9d81f32e585e1a985281e4ed473a7f635c895d78) as a demonstration of how I think you can fix the problems in the `KinesisSourceExecutor`.  Basically, it creates the clients explicitly.  It works because [the standard client uses the default region provider chain](https://github.com/aws/aws-sdk-java/blob/1.12.75/aws-java-sdk-kinesis/src/main/java/com/amazonaws/services/kinesis/AmazonKinesisClientBuilder.java#L41).  By contrast, the worker by default [uses us-east-1](https://github.com/awslabs/amazon-kinesis-client/blob/d8e29e8cd788d82b1de2c70a0db4ca74b207b632/src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java#L1571-L1572).

This is completely untested!!  But I've read through the sdk library and in theory this should work.  There's some tidy-up for you to do, e.g. configure the clients with region/endpoint.  And I think you can then also remove the region/endpoint settings from the `KinesisClientLibConfiguration`, although they don't harm.